### DISCUSSION TITLE
In-browser companion

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#366] [Ideas]
It would be neat if `dagger` could optionally launch a companion webapp to monitor and configure an environment. This companion webapp would not implement a specific feature, but rather provide a platform for delivering various features that are more appropriate for a web environment. This would allow the CLI to focus on what it does best, and relax the requirement to do absolutely everything in a POSIX terminal, which is very challenging.

In short: if `dagger` always has access to 2 modes of user interface, terminal and browser, it can deliver a better user experience than with only one of them.

Examples of products following this pattern:

* [Streamlit](https://streamlit.io)
* [VueCLI](https://cli.vuejs.org/)

Examples of possible in-browser features:
* Interactive configuration forms generated from cue spec. We had this feature in the Blocklayer SaaS beta: it is very nice!
* Visual DAG representation
* Interactive CUE sandbox
* Log streaming and search

**Comment by verdverm** [comment]
VueCLI is an excellent example of this

**Reply by shykes** [reply]
I will check it out, thanks!
